user/src/com/google/gwt/user/rebind/rpc/SerializableTypeOracleBuilder.java

```diff
@@ -869,6 +869,10 @@ public SerializableTypeOracle build(TreeLogger logger) throws UnableToCompleteEx
 
       if (tic.maybeEnhanced()
           || (enhancedClasses != null && enhancedClasses.contains(type.getQualifiedSourceName()))) {
+        logger.log(TreeLogger.WARN, "The class " + type.getQualifiedSourceName() + " is both " +
+                "referenced from configuration as rpc.enhancedClasses and has JPA annotations. " +
```

Comment:
The message should be `...has JPA/JDO annotations or is explicitly configured as enhanced class using configuration property ...`